### PR TITLE
KBC-1721 delete devBranch configuration

### PR DESCRIPTION
**Pull request by martinjunger**
Jira: KBC-1721

Before asking for review make sure that:

- [x] New client method(s) has tests
- [x] Apiary file is updated
- [x] You declared if there is a BC break or not (will affect next release of a client)

---

### CODE REVIEW

File: tests/Common/BranchComponentTest.php [hunk 1]
```diff
@@ -1004,6 +995,88 @@ public function testManipulationWithComponentConfigurations()
...
+        $this->assertCount(1, $mainConfigRows);
+
+        // delete soft-deleted dev branch configuration
+        try {
+            $branchComponents->deleteConfiguration($componentId, 'main-1');
+            $this->fail('Configuration should not be deleted in the dev branch');
+        } catch (ClientException $e) {
+            $this->assertSame(400, $e->getCode());
+            $this->assertSame('cannotDelete', $e->getStringCode());
+            $this->assertSame('In development branches is not posible delete configuration from trash.', $e->getMessage());
```
Comment (fejjo) [inline tests/Common/BranchComponentTest.php:1076]:
```suggestion
            $this->assertSame('In development branches is not possible delete configuration from trash.', $e->getMessage());
```

**Review by fejjo — CHANGES_REQUESTED** [review]

File: apiary.apib [hunk 2]
```diff
@@ -5668,6 +5668,26 @@ you have to delete it twice (call twice this API resource). The `canPurgeTrash`
 
 + Response 204
 
+### Delete Development Branch Configuration [DELETE /v2/storage/branch/{branch_id}/components/{component_id}/configs/{config_id}]
+Deletes a development branch configuration. When a configuration is deleted, it is only marked as deleted and remains still available in the system.
+You can [list deleted](#reference/component-configurations/component-configs/list-configs) configurations, or
+[restore deleted](#reference/component-configurations/manage-configs/restore-deleted-config) configurations.
```
[the remark this answers is not in the document]
Comment (martinjunger) [reply apiary.apib:5674]: Jasně, zapomněl jsem je upravit. Dík

[the remark this answers is not in the document]
Comment (martinjunger) [reply apiary.apib]: Opraveno. Ty reference jsou spatne uz cca 4 roky 🤔

**Review by fejjo — APPROVED** [review]
lgtm, este som pridal nieco do testov https://github.com/keboola/storage-api-php-client/pull/646/commits/084034accab98c2cf343e64693701f5e7b156aa6 ak ti to pride v pohode tak by som to asi squashol na nejake rozumne commity nech tam niesu tie fixupy a mergol. Nezabudni ze sa ma aj tagnut nova verzia klienta

**Pull request merged**